app/models/shift.rb

```diff
@@ -18,4 +18,25 @@ def self.today_driver_shifts(driver_id, date = nil)
     shifts = shifts_by_date(shifts_by_driver(Shift.all, driver_id), date)
     shifts.first
   end
+
+  def self.fill_month(templates, month_date)
+    current_shifts = shifts_for_month(month_date)
+
+    Shift.transaction do
+      (month_date.beginning_of_month..month_date.end_of_month).each do |date|
+        templates.where(day_of_week: date.wday).each do |template|
+          new_shift_attributes = { driver_id: template.driver_id, shift_type: template.shift_type, shift_date: date }
+
```

Comment:
is there really no way to do this than a triple nested loop *inside* a transaction?  also, if you are using `create!` you need to rescue errors and abort the transaction explicitly.